simexpal/schemes/experiments.json

```diff
@@ -431,16 +437,54 @@
 			"type": "array",
 			"items": {
 				"type": "object",
-				"required": ["axis", "items"],
-				"properties": {
-					"axis": {"type": "string"},
-					"items": {
-						"type": "array",
-						"items": {
-							"type": "object",
-							"required": ["name"],
+				"allOf": [
+					{
+						"if": {
+							"required": ["range", "steps"],
+							"properties": {
+								"range": {},
+								"steps": {}
+							}
+
+						},
+						"then": {
+							"required": ["axis", "range", "steps"],
+							"properties": {
+								"axis": {"type": "string"},
+								"extra_args": {"$ref":  "#/definitions/str_list"},
+								"range": {"$ref": "#/definitions/int_list"},
+								"steps": {"type": "integer"},
+								"num_nodes": {
+									"type": "integer",
+									"minimum": 1
+								},
+								"procs_per_node": {
+									"type":  "integer",
+									"minimum": 1
+								},
+								"num_threads": {
+									"type": "integer",
+									"minimum": 1
+								},
+								"environ": {"type": "object"},
+								"slurm_args": {"$ref": "#/definitions/str_or_str_list"},
+								"launcher": {"type": "string"}
```

Comment:
It's a bit ugly that we need to repeat these properties for all cases. Can't we handle that more concisely using another case in the `anyOf` or so?

Reply:
The best I could figure out, is to move out the definitions of the properties by adding the subschema:

``` JSON
"slurm_settings": {
	"type": "object",
	"properties": {
		"num_nodes": {
			"type": ["integer", "string"],
			"minimum": 1
		},
		"procs_per_node": {
			"type":  ["integer", "string"],
			"minimum": 1
		},
		"num_threads": {
			"type": ["integer", "string"],
			"minimum": 1
		},
               "slurm_args": {"$ref": "#/definitions/str_or_str_list"},
	}
}
```

and adding it to an ``allOf``:

``` JSON
"allOf": [
		{
			"$ref": "#/definitions/slurm_settings"
		},
		{
			"required": ["axis", "range", "steps"],
			"properties": {
				"axis": {"type": "string"},
				"extra_args": {"$ref":  "#/definitions/str_list"},
				"range": {"$ref": "#/definitions/int_list"},
				"steps": {"type": "integer"},
				"num_nodes": {},   // definition is in slurm_settings
				"procs_per_node": {},   // definition is in slurm_settings
				"num_threads": {},   // definition is in slurm_settings
                                "slurm_args": {},    // definition is in slurm_settings
				"environ": {"type": "object"},
				"launcher": {"type": "string"}
			},
			"additionalProperties": false
		}
	]
```

Note that we still have to list the properties in the second ``allOf`` term, as we are using ``"additionalProperties": false``. This would make the schema a bit more concise since with this PR we list those properties 4 times (with the same definition each time) in our schema.

Reply:
done as described above